File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const size = nums.length;
        if(size===0) return 0;
        const numSet = new Set(nums);

        let count=1;
        for(let num of numSet){
                if(!numSet.has(num-1)){
                    let current = num;
                    let  tempCount=1; 
                       while(numSet.has(current+1)){
                        tempCount++;
                        current ++;
                       }
                       count = Math.max(count,tempCount);
                }

        }
       return count;
    }
}

File: Data Structures & Algorithms/jump-game-ii/submission-1.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        
        farthest = 0
        end = 0
        jump = 0

        for i in range(len(nums)-1):
            farthest= max(farthest, i+nums[i])

            if i == end:
                end = farthest
                jump  += 1
        return jump